### ISSUE TITLE
type of parameter object changes after the first population

### ISSUE THREAD

**Issue by LiBuchauer** [ICB-DCM/pyABC#314]
Hi,

I noticed that the type of the parameter object which is passed into the model definition function during an ABCSMC run is not constant. To see what I mean, you can run the "Paramter inference" example and look at the type of ``parameter`` every time ``model()`` is called: 

```
def model(parameter):
    print(type(parameter))
    return {"data": parameter["mean"] + 0.5 * np.random.randn()}

prior = pyabc.Distribution(mean=pyabc.RV("uniform", 0, 5))

def distance(x, y):
    return abs(x["data"] - y["data"])

abc = pyabc.ABCSMC(model, prior, distance)

db_path = ("sqlite:///" +
           os.path.join(tempfile.gettempdir(), "test.db"))
observation = 2.5
abc.new(db_path, {"data": observation})

history = abc.run(minimum_epsilon=.1, max_nr_populations=10)
```

You will see that initially the type is `<class 'pyabc.parameters.Parameter'>`, but once the second population starts sampling, it becomes `<class 'pandas.core.series.Series'>`. 

I don't think is desired behaviour: at least for me it caused my model to crash as it relied on a dictionary method - I now have to handle it with an exception.

Best

Lisa

**Comment by yannikschaelte** [comment]
Hi Lisa, currently indeed the [`pyabc.transition.Transition`](https://github.com/ICB-DCM/pyABC/blob/master/pyabc/transition/base.py#L15)s return  [`pd.Series`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.html) objects, while the priors return [`pyabc.random_variable.Parameters`](https://github.com/ICB-DCM/pyABC/blob/master/pyabc/parameters.py#L30) objects. This is mainly due to efficiency, as the transitions operate on pandas objects anyway. 

I agree that this is not documented properly, we should at least change that. Basically, what we assume (and this could be changed) is that a "Parameter" is a sort-of dictionary object which allows accessing values via `par[key]`. Both `Parameters` and `Series` allow this. What additional functions do you need?

**Comment by LiBuchauer** [comment]
Hi Yannik, thanks for your reply. I agree that documentation would be the most important thing as it is kind of surprising. Whether to change the underlying cause it is probably a more philoshophical question.

In my specific case I need to normalise all parameters to sum to one within the model function, so assuming it was a dictionary I tried to call `parameters.values()` to then transform it into an np array. That fails for the series, but workarounds are available.

**Comment by yannikschaelte** [comment]
Dear @LiBuchauer , this should be fixed now in #318 -- the model now always gets a pyabc.Parameter object. Are you aware of any documentation that could be still improved?

**Comment by yannikschaelte** [comment]
fixed in #321
